phpBB/phpbb/content_visibility.php

```diff
@@ -144,7 +144,13 @@ public function get_count($mode, $data, $forum_id)
 	*/
 	public function is_visible($mode, $forum_id, $data)
 	{
-		$is_visible = $this->auth->acl_get('m_approve', $forum_id) || $data[$mode . '_visibility'] == ITEM_APPROVED;
+		$visibility = $data[$mode . '_visibility'];
+		$poster_key = ($mode === 'topic') ? 'topic_poster' : 'poster_id';
+		$is_visible = $this->auth->acl_get('m_approve', $forum_id) || $visibility == ITEM_APPROVED;
```

Comment:
to optimize this use
``$visibility == ITEM_APPROVED || $this->auth->acl_get('m_approve', $forum_id)``